lib/parser/gpx.rb

```diff
@@ -42,6 +46,40 @@ def parse # rubocop:disable Metrics/AbcSize
 
     private
 
+    def enhance_trace(points)
+      prev_point = points.first
+      points[1..].each_with_object([]) do |cur_point, et|
+        duration = cur_point[:time].to_f - prev_point[:time].to_f
+        
+        prev = Geokit::LatLng.new(prev_point[:lat].to_f, prev_point[:lon].to_f)
+        cur  = Geokit::LatLng.new(cur_point[:lat].to_f, cur_point[:lon].to_f)
+        distance = prev.distance_to(cur)
+        
+        speed = distance / duration / 1000 * 3600
+        
+        et << cur_point.merge(distance: distance, duration: duration, speed: speed)
+        prev_point = cur_point
+      end
+    end
+
+    def set_thresholds(points)
+      # calculate average speed
+      # and assume to slow is 1/10th of it
+      avg_speed = points.sum { |p| p[:speed] } / points.size
+      @to_slow = avg_speed / 10
+
+      # get the average durations between the gps points
+      # and remove the lower and upper 10% to remove the
```

Comment:
Is there a reason for 10%? (just curious ☺️)

Reply:
I just wanted to get rid of the highest and lowest values.. any other suggestion ?